### PR TITLE
Tables documentation

### LINKED ISSUE #2964: [Caloris] [Structured Data] Public API documentation

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes https://github.com/dust-tt/dust/issues/2964

Introduce Tables documentation. at least 75% of it.

## Risk

N/A

## Deploy Plan

- deploy `docs`
- deploy `front`

### CODE REVIEW

File: types/src/front/lib/core_api.ts [hunk 1]
```diff
@@ -96,18 +96,19 @@ type GetDatasetsResponse = {
   datasets: { [key: string]: CoreAPIDatasetVersion[] };
 };
 
-export type CoreAPITableSchema = Record<
-  string,
-  "int" | "float" | "text" | "bool" | "datetime"
->;
+export type CoreAPITableSchema = {
+  name: string;
+  value_type: "int" | "float" | "text" | "bool" | "datetime";
```
Comment (spolu) [inline types/src/front/lib/core_api.ts:101]: @fontanierh are these still up to date (I'm looking at datetime)

File: docs/src/pages/tables.mdx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    ### JSON body attributes
+
+    Attributes are passed as a JSON object in the request body.
+
+    <Properties>
+      <Property name="name" type="string">
+        The name of the Table to create (must be unique within a Data Source).
+      </Property>
+      <Property name="description" type="string">
+        The description of the Table (used by the model to generate SQL query).
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:86]:
```suggestion
        The description of the Table (used by the model to generate SQL queries).
```

File: docs/src/pages/tables.mdx [hunk 3]
```diff
@@ -0,0 +1,336 @@
+export const description =
+  "On this page, we’ll dive into the Tables endpoints you can use to manage tables programmatically.";
+
+# Tables
+
+On this page we'll dive into the Tables endpoint you can use to manage Data Sources' Tables
+programmatically. Tables let you insert tabular data that can be queried by an assistant using the
+`Query Tables` action (Generates a SQL query that is then run on your tabular data and whose results
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:8]:
```suggestion
`Query Tables` action (Generates a SQL query that is then ran on your tabular data and whose results
```

File: docs/src/pages/tables.mdx [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+<Properties>
+  <Property name="table_id" type="string">
+    The table ID as specified or generated at creation.
+  </Property>
+  <Property name="table_name" type="string">
+    The table name as specified at creation.
+  </Property>
+  <Property name="description" type="string">
+    The description of the table (used by the model to generate SQL query).
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:49]:
```suggestion
    The description of the table (used by the model to generate SQL queries).
```

File: docs/src/pages/tables.mdx [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+
+---
+
+## Create Rows {{ tag: 'POST', label: '/v1/w/:workspace_id/data_sources/:data_source_name/tables/:table_id/rows' }}
+
+<Row>
+  <Col>
+
+    This endpoint enables you to add rows to an existing Table. Rows are JSON objects associating
+    keys to values. The keys and value types (string, number, boolean) must match the Table's
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:190]: we also support datetime: https://github.com/dust-tt/dust/blob/82c5ca353a31b546b64ee33b9d1fc264db313eb6/front/pages/api/v1/w/%5BwId%5D/data_sources/%5Bname%5D/tables/%5BtId%5D/rows/index.ts#L24-L27

File: docs/src/pages/tables.mdx [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+</Row>
+
+---
+
+## Create Rows {{ tag: 'POST', label: '/v1/w/:workspace_id/data_sources/:data_source_name/tables/:table_id/rows' }}
+
+<Row>
+  <Col>
+
+    This endpoint enables you to add rows to an existing Table. Rows are JSON objects associating
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:189]: Also allows you to update rows. It's more `Upsert Rows` than it is `Create Rows`

File: docs/src/pages/tables.mdx [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+      </Property>
+    </Properties>
+
+    ### JSON body attributes
+
+    Attributes are passed as a JSON object in the request body.
+
+    <Properties>
+      <Property name="rows" type="Row[]">
+        The rows to insert.
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:228]: to upsert

File: docs/src/pages/tables.mdx [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+      </Property>
+    </Properties>
+
+    ### Optional JSON body attributes
+
+    <Properties>
+      <Property name="truncate" type="boolean">
+        If true the content of the Table is erased and replaced by the new rows. Otherwise rows
+        are appended to the Table.
```
Comment (fontanierh) [inline docs/src/pages/tables.mdx:237]: Or updated

**Review by fontanierh — CHANGES_REQUESTED** [review]
Thank you !! 🙏 

Just a small change request wrt the `/rows` endpoint which is actually an `upsert` endpoint

File: docs/src/pages/tables.mdx (hunk 3 above)
Comment (spolu) [reply docs/src/pages/tables.mdx:8]: uh nope

File: docs/src/pages/tables.mdx (hunk 2 above)
Comment (spolu) [reply docs/src/pages/tables.mdx:86]: :+1:

File: docs/src/pages/tables.mdx (hunk 4 above)
Comment (spolu) [reply docs/src/pages/tables.mdx:49]: :+1:

File: docs/src/pages/tables.mdx (hunk 3 above)
Comment (fontanierh) [reply docs/src/pages/tables.mdx:8]: My bad 🙈

File: types/src/front/lib/core_api.ts (hunk 1 above)
Comment (fontanierh) [reply types/src/front/lib/core_api.ts:101]: Yes datetime is still supported

File: docs/src/pages/tables.mdx (hunk 5 above)
Comment (spolu) [reply docs/src/pages/tables.mdx]: :+1:

File: docs/src/pages/tables.mdx (hunk 6 above)
Comment (spolu) [reply docs/src/pages/tables.mdx]: :+1:

File: docs/src/pages/tables.mdx (hunk 7 above)
Comment (spolu) [reply docs/src/pages/tables.mdx]: :+1:

File: docs/src/pages/tables.mdx (hunk 8 above)
Comment (spolu) [reply docs/src/pages/tables.mdx]: :+1:

File: docs/src/pages/tables.mdx (hunk 3 above)
Comment (spolu) [reply docs/src/pages/tables.mdx:8]: np!

**Comment by spolu** [comment]
Thanks @fontanierh for the suggestions :+1:

**Pull request merged**